packages/search/src/test/use-fuzzy-search.tsx

```diff
@@ -0,0 +1,63 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import Search, { useFuzzySearch } from '..';
+import type { UseFuzzySearchOptions } from '../use-fuzzy-search';
+
+const TestComponent = < T, >( props: UseFuzzySearchOptions< T > ) => {
```

Comment:
Out of curiosity (I'm new to TypeScript), what does `< T, >` designate?

Reply:
I just learnt this trick yesterday myself! The short story is that it means the same thing as `< T >` because the trailing slash will be ignored. So it's signifying that this anonymous function is [generic](https://www.typescriptlang.org/docs/handbook/2/generics.html).

The reason for the comma is that it helps work around a slight incompatibility between the TypeScript and JSX languages. When the JSX parser sees `< T >` it thinks it's a React element. When the TypeScript parser sees `< T >` it thinks it signifies some sort generic. Usually it's possible to disambiguate between the two:

```
// These couldn't possibly be React elements because of their context
function foo< A >....
const m = new Set< number >...

// But what about this one ....
const a = < B >...
```

The rule is that the last example will always be a React element and will never represent something like a generic anonymous function (in `.tsx` files at least, I think the rules are different in `.ts` files). You can disambiguate for the parser by doing something like `< B, >` or `< B extends Sometype >`. Now it knows it must TypeScript syntax.

Anyway, that's why in `.tsx` files I think it's a lot more common to see non-anonymous `function` functions. There are fewer weird rules. It's also why the TypeScript compiler _insists_ that JSX can only be written in `.tsx` files, unlike Babel which will accept JSX syntax in `.js` files.

Reply:
👍 Makes sense. Thanks for the explanation, @p-jackson!